native/src/main/java/io/ballerina/stdlib/data/jsondata/FromString.java

```diff
@@ -0,0 +1,188 @@
+        add(PredefinedTypes.TYPE_DECIMAL);
+        add(PredefinedTypes.TYPE_STRING);
+    }};
+    private static final UnionType JSON_TYPE_WITH_BASIC_TYPES = TypeCreator.createUnionType(BASIC_JSON_MEMBER_TYPES);
+
+    public static Object fromStringWithType(BString string, BTypedesc typed) {
+        Type expType = typed.getDescribingType();
+
+        try {
+            return fromStringWithType(string, expType);
+        } catch (NumberFormatException e) {
+            return returnError(string.getValue(), expType.toString());
+        }
+    }
+
+    public static Object fromStringWithType(BString string, Type expType) {
+        String value = string.getValue();
+        try {
+            switch (expType.getTag()) {
+                case TypeTags.INT_TAG:
+                    return stringToInt(value);
+                case TypeTags.FLOAT_TAG:
+                    return stringToFloat(value);
+                case TypeTags.DECIMAL_TAG:
+                    return stringToDecimal(value);
+                case TypeTags.STRING_TAG:
+                    return string;
+                case TypeTags.BOOLEAN_TAG:
+                    return stringToBoolean(value);
+                case TypeTags.NULL_TAG:
+                    return stringToNull(value);
+                case TypeTags.UNION_TAG:
+                    return stringToUnion(string, (UnionType) expType);
+                case TypeTags.JSON_TAG:
+                    return stringToUnion(string, JSON_TYPE_WITH_BASIC_TYPES);
+                case TypeTags.TYPE_REFERENCED_TYPE_TAG:
+                    return fromStringWithType(string, ((ReferenceType) expType).getReferredType());
+                default:
+                    return returnError(value, expType.toString());
+            }
+        } catch (NumberFormatException e) {
+            return returnError(value, expType.toString());
+        }
+    }
+
+    private static Long stringToInt(String value) throws NumberFormatException {
+        return Long.parseLong(value);
+    }
+
+    private static Double stringToFloat(String value) throws NumberFormatException {
+        if (hasFloatOrDecimalLiteralSuffix(value)) {
+            throw new NumberFormatException();
+        }
+        return Double.parseDouble(value);
+    }
+
+    private static BDecimal stringToDecimal(String value) throws NumberFormatException {
+        return ValueCreator.createDecimalValue(value);
+    }
+
+    private static Object stringToBoolean(String value) throws NumberFormatException {
+        if ("true".equalsIgnoreCase(value) || "1".equalsIgnoreCase(value)) {
+            return true;
+        }
+
+        if ("false".equalsIgnoreCase(value) || "0".equalsIgnoreCase(value)) {
+            return false;
+        }
+        return returnError(value, "boolean");
+    }
+
+    private static Object stringToNull(String value) throws NumberFormatException {
+        if ("null".equalsIgnoreCase(value) || "()".equalsIgnoreCase(value)) {
```

Comment:
I think same thing goes in here has well. I Json a null value can only be `null`.